client.go

```diff
@@ -189,12 +189,19 @@ func (d BaseClient) doRequest(httpRequest *http.Request, target any, opts ...Opt
 		if d.config.Non2xxStatusHandler != nil {
 			d.config.Non2xxStatusHandler(httpResponse, data)
 		}
+		// Checking for OperationOutcome
```

Comment:
Nitpick: This comment doesn't add any information not provided by the method name in the next line